pulseeffects: 4.8.4 -> 5.0.0, pulseeffects-legacy: init at 4.8.4

ymeister opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Version 5.0.0 released:
https://github.com/wwmm/pulseeffects/releases/tag/v5.0.0

As of this version PulseEffects uses PipeWire instead of PulseAudio. Because of the currently lesser adoption of PipeWire, legacy version of PulseEffects (pulseeffects-legacy) was added. Further updates of the legacy version can be acquired from corresponding branch https://github.com/wwmm/pulseeffects/tree/pulseaudio-legacy. Please note, that legacy branch is in maintenance mode and no longer under development.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

SuperSandro2000 commented on pkgs/applications/audio/pulseeffects/default.nix:
```diff
@@ -45,13 +45,13 @@ let
   ];
 in stdenv.mkDerivation rec {
   pname = "pulseeffects";
-  version = "4.8.4";
```
Please put this into a separate commit.

SuperSandro2000 commented on pkgs/applications/audio/pulseeffects-legacy/default.nix:
```diff
@@ -0,0 +1,115 @@
+  meta = with lib; {
+    description = "Limiter, compressor, reverberation, equalizer and auto volume effects for Pulseaudio applications";
+    homepage = "https://github.com/wwmm/pulseeffects";
+    license = licenses.gpl3;
```
gpl3Plus or gpl3Only?

ymeister replied:
I'm not familiar with this topic. Could you please check yourself? (Don't want to accidentally make any mistakes)
https://github.com/wwmm/pulseeffects/blob/master/LICENSE.md

rmcgibbo replied:
https://github.com/wwmm/pulseeffects/blob/1bb9be071bcf0c1f89b67a7e4eb8d3405dcb8a18/include/autogain.hpp#L9 looks like gpl3Plus

ymeister replied:
So be it.

r-rmcgibbo commented:
Result of `nixpkgs-review pr 112591` at f979af82 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>pulseeffects</li>
    <li>pulseeffects-legacy</li>
  </ul>
</details>

r-rmcgibbo commented:
Result of `nixpkgs-review pr 112591` at f979af82 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/eb49b59eefaf4b2eed0a8f66f50b24c0">pulseeffects</a></li></li>
    <li><a href="https://gist.github.com/550a42092fde3667c6a22ae491f9623a">pulseeffects-legacy</a></li></li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/applications/audio/pulseeffects-legacy/default.nix:110:5:
```
    |
110 |     license = licenses.gpl3;
    |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/applications/audio/pulseeffects/default.nix:110:5:
```
    |
110 |     license = licenses.gpl3;
    |     ^
```
</li>
  </ul>
</details>

jtojnar commented:
Does this still need `gst-plugins-good` now that it switched from `pulsesrc` to `pipewiresrc`? Could you to remove it, test pulseeffects and check the output in the terminal for any messages about missing GStreamer elements?

ymeister commented:
> Does this still need `gst-plugins-good` now that it switched from `pulsesrc` to `pipewiresrc`? Could you try to remove it, test pulseeffects and check the output in the terminal for any messages about missing GStreamer elements?

Still produces an error, although in slightly different form:
```
(pulseeffects:5480): glibmm-ERROR **: 06:28:23.639:
unhandled exception (type std::exception) in signal handler:
what: soe: Failed to get required plugin: spectrum

Trace/breakpoint trap (core dumped)
```

jtojnar commented:
Looks like `spectrum` is [part of the good plug-ins](https://gstreamer.freedesktop.org/documentation/spectrum/index.html?gi-language=c#factory-details). Could you update the comment accordingly?

ymeister committed:
pulseeffects: 4.8.4 -> 5.0.0

ymeister committed:
pulseeffects-legacy: init at 4.8.4

jtojnar reviewed (approved):
Changes look good to me but I cannot test it with pipewire-pulse at the moment.

jtojnar committed:
pulseeffects: Deprecate the old name

pulseeffects no longer supports PulseAudio.
Let’s move it to a new attribute name to avoid the breakage.

jtojnar commented:
After IRC discussion, I have decided to deprecate the old attribute name to avoid people being confused why it crashes: see https://github.com/wwmm/pulseeffects/issues/891 and https://github.com/wwmm/pulseeffects/issues/874
